libgnucash/engine/Account.cpp

```diff
@@ -2691,6 +2691,45 @@ DxaccAccountSetCurrency (Account * acc, gnc_commodity * currency)
 /********************************************************************\
 \********************************************************************/
 
+static gpointer
+account_foreach_descendant (const Account *acc, AccountCb2 thunk,
+                            gpointer user_data, gboolean sort, gboolean can_abort)
+{
+    const AccountPrivate *priv;
+    gpointer result = nullptr;
+    GList *children;
+
+    g_return_val_if_fail (GNC_IS_ACCOUNT(acc), nullptr);
+    g_return_val_if_fail (thunk, nullptr);
+
+    priv = GET_PRIVATE(acc);
```

Comment:
In C++ we declare at first use except when first use is inside a scope and we'll need it outside of the scope later and only specify the type when required.  Prefer brace initialization and reserving `=` for assignment.

`auto priv{GET_PRIVATE(acc)};`